### PR TITLE
Fix compat for hsa_rocr_jll

### PR DESCRIPTION
**Pull request by pxl-th**
This PR fixes compat to unblock https://github.com/JuliaRegistries/General/pull/67535

### CODE REVIEW

**Comment by giordano** [comment]
This is touching lots of files

**Comment by pxl-th** [comment]
> This is touching lots of files

Should be good now.

File: H/hsa_rocr_jll/Compat.toml [hunk 1]
```diff
@@ -7,9 +7,10 @@ JLLWrappers = "1.2.0-1"
...
 
-["4.2-4"]
+["4.2"]
 julia = "1.7.0-1"
+Elfutils_jll = "0.182"
 
 ["4.5-4"]
-Elfutils_jll = "0.182"
 julia = "1.8.0-1"
+Elfutils_jll = "0.182"
```
Comment (giordano) [inline H/hsa_rocr_jll/Compat.toml:16]: Am I correct to say this is unnecessary? I'm struggling to see the difference.  Also, isn't this is the same version of `Elfutils_jll` as above?

Comment (pxl-th) [reply H/hsa_rocr_jll/Compat.toml]: Yes, reverted.

**Pull request merged**